src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshalling/Utf8StringMarshaller.cs

```diff
@@ -101,5 +103,26 @@ public void FreeNative()
             if (_allocated != null)
                 Marshal.FreeCoTaskMem((IntPtr)_allocated);
         }
+
+        internal static string? ConvertToManaged(byte* value)
```

Comment:
If these internal helpers are a sore spot, I can try to update the built-in marshalling to use the public methods and follow the regular pattern that depends on the marshaller instance everywhere.

Reply:
I think that is the sore spot for me, at least that is what I initially noticed. If we could get rid of them I think I'd be willing to let my other concerns go.

Reply:
Ok, I have reverted the internal helpers and change the IL generation to target the public methods. How does it look?